lib/rubber/recipes/rubber/instances.rb

```diff
@@ -262,20 +262,30 @@ def create_instances(instance_aliases, instance_roles, create_spot_instance=fals
   def create_instance(instance_alias, instance_roles, create_spot_instance)
     role_names = instance_roles.collect{|x| x.name}
     env = rubber_cfg.environment.bind(role_names, instance_alias)
+    cloud_env = env.cloud_providers[env.cloud_provider]
 
-    monitor.synchronize do
-      cloud.before_create_instance(instance_alias, role_names)
-    end
+    availability_zone = cloud_env.availability_zone
 
     security_groups = get_assigned_security_groups(instance_alias, role_names)
 
-    cloud_env = env.cloud_providers[env.cloud_provider]
     ami = cloud_env.image_id
     ami_type = cloud_env.image_type
-    availability_zone = cloud_env.availability_zone
     region = cloud_env.region
     fog_options = cloud_env.fog_options || {}
 
+    instance_item = Rubber::Configuration::InstanceItem.new(instance_alias, env.domain, instance_roles, nil, ami_type, ami, security_groups)
+
+    if cloud_env.vpc_alias
```

Comment:
If you pass the entire `instance_item` through to `before_create_instance`, I think the VPC provider can handle this setup and you can thus remove it from the general `create_instance` method.